lti_consumer/tests/unit/test_lti_consumer.py

```diff
@@ -0,0 +1,664 @@
+        """
+        self.xblock.has_score = False
+        self.xblock.weight = 1.0
+
+        self.assertIsNone(self.xblock.max_score())
+
+
+class TestSetScore(TestLtiConsumerXBlock):
+    """
+    Unit tests for LtiConsumerXBlock.set_user_module_score() and LtiConsumerXBlock.clear_user_module_score()
+    """
+
+    def test_rebind_called(self):
+        """
+        Test that `runtime.rebind_noauth_module_to_user` is called
+        """
+        user = Mock(user_id=FAKE_USER_ID)
+        self.xblock.set_user_module_score(user, 0.92, 1.0, 'Great Job!')
+
+        self.xblock.runtime.rebind_noauth_module_to_user.assert_called_with(self.xblock, user)
+
+    def test_publish_grade_event_called(self):
+        """
+        Test that `runtime.publish` is called
+        """
+        user = Mock(id=FAKE_USER_ID)
+        score = 0.92
+        max_score = 1.0
+        self.xblock.set_user_module_score(user, score, max_score)
+
+        self.xblock.runtime.publish.assert_called_with(self.xblock, 'grade', {
+            'value': score,
+            'max_value': max_score,
+            'user_id': FAKE_USER_ID
+        })
+
+    def test_score_is_none(self):
+        """
+        Test when score parameter is None
+        """
+        max_score = 1.0
+        user = Mock(id=FAKE_USER_ID)
+        self.xblock.set_user_module_score(user, None, max_score)
+
+        self.assertEqual(self.xblock.module_score, None)
+
+    def test_max_score_is_none(self):
+        """
+        Test when max_score parameter is None
+        """
+        user = Mock(id=FAKE_USER_ID)
+        self.xblock.set_user_module_score(user, 0.92, None)
+
+        self.assertEqual(self.xblock.module_score, None)
+
+    def test_score_and_max_score_populated(self):
+        """
+        Test when both score and max_score parameters are not None
+        """
+        user = Mock(id=FAKE_USER_ID)
+        score = 0.92
+        max_score = 1.0
+        self.xblock.set_user_module_score(user, score, max_score)
+
+        self.assertEqual(self.xblock.module_score, score * max_score)
+
+    def test_no_comment_param(self):
+        """
+        Test when no comment parameter is passed
+        """
+        self.xblock.set_user_module_score(Mock(), 0.92, 1.0)
+
+        self.assertEqual(self.xblock.score_comment, '')
+
+    def test_comment_param(self):
+        """
+        Test when comment parameter is passed
+        """
+        comment = 'Great Job!'
+        self.xblock.set_user_module_score(Mock(), 0.92, 1.0, comment)
+
+        self.assertEqual(self.xblock.score_comment, comment)
+
+    @patch('lti_consumer.LtiConsumerXBlock.set_user_module_score')
+    def test_clear_user_module_score(self, mock_set_user_module_score):
+        """
+        Test that clear_user_module_score calls set_user_module_score with params set to None
+        """
+        user = Mock()
+        self.xblock.clear_user_module_score(user)
+        mock_set_user_module_score.assert_called_with(user, None, None)
+
```

Comment:
@doctoryes Yes, there are. Check out the tests in [test_lti.py](https://github.com/edx/xblock-lti-consumer/pull/1/files#diff-203113196545de710258838493172076R86) and [test_outcomes.py](https://github.com/edx/xblock-lti-consumer/pull/1/files#diff-a0dbcbf039b96fab3e0e5f822c260f65R210).